src/main/java/org/apache/sling/resourcemerger/impl/MergedResourcePickerWhiteboard.java

```diff
@@ -64,7 +67,11 @@ protected void deactivate() {
     }
 
     @Reference(policy = ReferencePolicy.DYNAMIC, cardinality = ReferenceCardinality.MULTIPLE)
-    public void bindMergedResourcePicker(MergedResourcePicker resourcePicker, Map<String, Object> properties) {
+    public void bindMergedResourcePicker(MergedResourcePicker resourcePicker,
```

Comment:
@raducotescu TBH I don't understand this. Why are the defaults not injected in the bind method?
The spec says
> An event method can take one or more parameters.An event method can take one or more parameters. ...
> Map - An unmodifiable Map containing the service properties of the bound service. This Map must additionally implement Comparable with the compareTo method comparing service property maps using the same ordering as ServiceReference.compareTo based upon service ranking and service id.
(https://docs.osgi.org/specification/osgi.cmpn/8.0.0/service.component.html#service.component-method.injection)

Reply:
I think it had something to do with the change from #8 not being implemented the same way in my initial set of changes. You're right - if I compare the properties map from the service reference with the map parameter of the previous signature they are now identical.

I can revert the commit for SLING-11774 without any problems.

Reply:
See https://github.com/apache/sling-org-apache-sling-resourcemerger/commit/a04dcc3048cb6c9d719e41a5a04593b6942fdb92.